internal/storage/debug_bucket.go

```diff
@@ -126,6 +126,16 @@ func (b *debugBucket) Name() string {
 }
 
 func (b *debugBucket) BucketType() gcs.BucketType {
+	var err error
+
+	// BucketType internally calls  GetStorageLayout API from control client , hence
+	// logging this as a GetStorageLayout call.
```

Comment:
Will the call be logged even if GetStorageLayout GCS call has not happened ? BucketType() only calls the api once and uses the type throughout the lifecycle of mount

Reply:
Yes. Even when the actual network call has not happened, this is logged as a GCS call (time of request completion is considerably less).
Another scenario I am unsure about is when the first call comes in for a non hierarchical bucket with nil control client, 
```
if bh.bucketType == gcs.Nil {
		if bh.controlClient == nilControlClient {
			bh.bucketType = gcs.NonHierarchical
			return bh.bucketType
		}
```
then GetStorageLayout api is not called, but this is also logged with minimal time taken for serving request.

Reply:
then the logging information is not correct, the log should only come when gcs call happens

Reply:
Resolving since as per offline discussion, logging logic is now in bucket_handle.go , so , this issue is taken care of.